### ISSUE TITLE
[Potential Bug/Question] Bottom share..always button placement inconsistency

### ISSUE THREAD

**Issue by Gravemind2015** [1fexd/LinkSheet#41]
Version: 0.0.29 from Izzyondroid(cleared data afterwards for testing)

When a new link is clicked, the copy/share/once/always button appear on bottom (nice idea IMO), and is consistently on bottom if 'Never show last picked app' is disabled.

However if 'Never show last picked app' is enabled, then these buttons show at the bottom for *new* links, but once an app is selected, it appears on top and these buttons show up below it, above other apps like in previous releases.

Attached below are two screen recordings showing the behaviour and inconsistency in position. Note that it happens in both list and grid.

https://imgur.com/a/fHhL3kq
https://imgur.com/a/aQVfLcl

I wonder if this is the intended behaviour when last picked app is enabled, and for new links since there's no last picked app, the grid/list is shown and buttons are on bottom. Or it's a bug indeed and these buttons are supposed to be always at bottom.

**Comment by Gravemind2015** [comment]
> > When a new link is clicked, the copy/share/once/always button appear on bottom (nice idea IMO), and is consistently on bottom if 'Never show last picked app' is disabled.
> > However if 'Never show last picked app' is enabled, then these buttons show at the bottom for _new_ links, but once an app is selected, it appears on top and these buttons show up below it, above other apps like in previous releases.
> 
> That is not the behavior shown in the recordings you provided, at least from what I could tell:
> 
>     * Recording 1: "Never show 'last picked app' section" disabled. You switch between a new link and an opened link, and as expected, the previously opened link shows the last picked app in the top section, and when you choose a new link, the buttons are moved down to the bottom, as the top section no longer exists.
> 
>     * Recording 2: "Never show 'last picked app' section" is enabled. You select a previously opened link, which as expected, does not show the last picked app. As the top section doesn't exist, buttons are at the bottom. You then disable "Never show 'last picked app' section" and select two previously opened links, which both show the top section with the buttons, as expected.
> 
> 
> Am I missing something here?

The links were not sorted as per the description, my bad. I'd also clipped the 2nd recording which started with that option disabled, in retrospect it looks backwards w.r.t my description now.

What I was confused about is whether the buttons following the last app section (if exists) is intentional or are they supposed to always be at the bottom regardless of last app section existing and thus this being a glitch.

The reasons for this confusion are some recent suggestions in other posts about keeping the recently accessed app at the bottom or pinned to a specific place. If dev is planning to have buttons consistently on the bottom of entire sheet, then it's not behaving that way (and if it's intentional then obviously I'm idiot haha).

**Comment by Gravemind2015** [comment]
Yeah, I understand. I don't intend to use the last app part, just grid view and everything stays consistent, with buttons on bottom so they don't interfere. 

About that 'always' button, it is really a pain to revert. There's even no 'clear defaults' button on my samsung anymore, last time I had to install some random browser, set it as default and uninstall it to clear my default browser selection. 

The only reason to report was to figure out whether anchoring buttons to bottom was the intention as the option to disable that last app part was added in the recent release. Otherwise the current behaviour also makes sense in 'aosp default behaviour' as well IMO.

**Comment by 1fexd** [comment]
I don't quite understand what the issue is here. Can you explain it again @Gravemind2015

**Comment by Gravemind2015** [comment]
@1fexd sorry, I'll try to explain again. There were some quirks from my phone that made a debug version of linksheet somewhat buggy, leading me to believe some of it's behaviours were intentional. I've since tested clean setup on another phone and found what I reported as a 'bug' was probably the intended behaviour. Anyway, this still can serve as a suggestion, so I'll try to explain better.

I was talking about the behaviour of the Copy to/Share to/Just Once/Always buttons when the 'last picked app' option is enabled. 

1. Previously there was no toggle to turn the last picked app section off. 
2. When enabled, clicking on any new domain/link, shows a list/grid and since no 'last' app is there, that section doesn't appear on top. All these action buttons are at the bottom of screen.
4. When a user clicks on a link/domain that has been accessed before, the 'last' app part shows up at the top. Now the buttons move up, below the last used app but above the rest of apps.
5. This has been the standard behaviour since the start, but what I had assumed as a change (it was a weird glitch) was that the buttons were supposed to be at the absolute bottom at all times, i.e not attached to the last used app.

Again, I apologise for the mishap and for wasting everyone's time here, but the ensuing discussion led to an interesting observation, the movement of these buttons and their placement between the last used app and other apps can lead to mis-clicks. Maybe they should be always at the very bottom? That could be a potentially more consistent way to display them.
